src/vm/tests/sequences.rs

```diff
@@ -35,6 +35,176 @@ fn test_simple_list_admission() {
     });
 }
 
+#[test]
+fn test_string_ascii_admission() {
+    let defines =
+        "(define-private (set-name (x (string-ascii 11))) x)";
+    let t1 = format!("{} (set-name \"hello world\")", defines);    
+
+    let expected = Value::string_ascii_from_bytes("hello world".into()).unwrap();
+
+    assert_eq!(expected, execute(&t1).unwrap().unwrap());
+}
+
+#[test]
+fn test_string_utf8_admission() {
+    let defines =
+        "(define-private (set-name (x (string-utf8 14))) x)";
+    let t1 = format!("{} (set-name u\"my 2 \\u{{c2a2}} (cents)\")", defines);    
+
+    let expected = Value::string_utf8_from_string_utf8_literal("my 2 \\u{c2a2} (cents)".into()).unwrap();
+
+    assert_eq!(expected, execute(&t1).unwrap().unwrap());
+}
+
+#[test]
+fn test_string_ascii_map() {
+    let defines =
+        "(define-private (replace-a-with-b (c (string-ascii 1))) (if (is-eq \"a\" c) \"b\" c))";
+    let t1 = format!("{} (map replace-a-with-b \"ababab\")", defines);    
+
+    let expected = Value::list_from(vec![
+        Value::string_ascii_from_bytes("b".into()).unwrap(),
+        Value::string_ascii_from_bytes("b".into()).unwrap(),
+        Value::string_ascii_from_bytes("b".into()).unwrap(),
+        Value::string_ascii_from_bytes("b".into()).unwrap(),
+        Value::string_ascii_from_bytes("b".into()).unwrap(),
+        Value::string_ascii_from_bytes("b".into()).unwrap(),
+        ]).unwrap();
+
+    assert_eq!(expected, execute(&t1).unwrap().unwrap());
+}
+
+#[test]
+fn test_string_utf8_map() {
+    let defines =
+        "(define-private (replace-dog-with-fox (c (string-utf8 1))) (if (is-eq u\"\\u{1F436}\" c) u\"\\u{1F98A}\" c))";
+    let t1 = format!("{} (map replace-dog-with-fox u\"fox \\u{{1F436}}\")", defines);    
+
+    let expected = Value::list_from(vec![
+        Value::string_utf8_from_bytes("f".into()).unwrap(),
+        Value::string_utf8_from_bytes("o".into()).unwrap(),
+        Value::string_utf8_from_bytes("x".into()).unwrap(),
+        Value::string_utf8_from_bytes(" ".into()).unwrap(),
+        Value::string_utf8_from_bytes("🦊".into()).unwrap(),
```

Comment:
Emojis in code. The future has arrived.

Reply:
Lol, yeah this test is highly controversial, but it's a good "end to end" test. Happy to take another approach if we want to avoid this.